### DISCUSSION TITLE
How can one specify the dependent variable as likelihood (or conditional likelihood) in nlmixr2?

### DISCUSSION THREAD

**Discussion by Chenguang-W** [nlmixr2/nlmixr2#202] [Q&A]
Hello Matt, 

From examples provided in the document, I only found dependent variable (Y) defined as PK concentration or PD markers such as CIRC ~ prop(prop.err). Is it possible to specify the Y as the likelihood function for categorical model or time-to-event model? I think the nmlxr2 has all the elements (algorithm, solver, etc.) to fit such kind of models, but don't know how to specify it. Any example code would be appreciated.

**Comment by mattfidler** [comment]
Hi @GDPOG,

This is  a newer feature (and only works with `focei` and the population only solvers like `nlm` that were just released).

I don't have any code handy but this is discussed here:

https://nlmixr2.github.io/rxode2/articles/rxode2-syntax.html#residual-functions-when-using-rxode2-functions


In particular this discusses ordinal likelihood

https://nlmixr2.github.io/rxode2/articles/rxode2-syntax.html#ordinal-likelihoods

And this talks about using `ll(cmt)` for log-likelihood

https://nlmixr2.github.io/rxode2/articles/rxode2-syntax.html#distributions-of-known-likelihoods

Hopefully this helps.

**Reply by mattfidler** [reply]
For specified distributions  like `binom()` it also simulates the type of input data (same with categorical).

There is not a time to event model yet because the simulation been worked out yet (though the likelihood can be specified by `ll(cmt)`)
